File: src/main/js/models/nodes/NumberNode.ts
import { Node } from "./Node";
import { NodeType } from "./NodeType";

/**
 * Number node is a wrapper for a number.
 */
export class NumberNode extends Object implements Node {
  readonly type: NodeType = NodeType.Number;
  readonly value: number;

  constructor(value: number) {
    super();
    this.value = value;
  }

  toString() {
    return this.value.toString();
  }
}

File: src/main/java/io/protobase/f7/models/NumberNode.java
package io.protobase.f7.models;

import com.google.common.base.MoreObjects;

/**
 * Number node is a wrapper for a number as a Double.
 */
public class NumberNode extends BaseObject implements Node {
  private Double value;

  /**
   * New node from double.
   *
   * @param value to wrap.
   */
  public NumberNode(Double value) {
    this.value = value;
  }

  /**
   * Get the value.
   *
   * @return double.
   */
  public Double getValue() {
    return value;
  }

  @Override
  public String toString() {
    return MoreObjects.toStringHelper(this)
        .add("value", value)
        .toString();
  }

  @Override
  public Object[] significantAttributes() {
    return new Object[]{
        value
    };
  }
}
